twap/main.go

```diff
@@ -0,0 +1,219 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/blocky/as-demo/as"
+	"github.com/blocky/as-demo/price"
+)
+
+type ArgsIterate struct {
+	TokenAddress string          `json:"token_address"`
+	ChainID      string          `json:"chain_id"`
+	NumSamples   int             `json:"num_samples"`
+	EAttest      json.RawMessage `json:"eAttest"`
+	TAttest      json.RawMessage `json:"tAttest"`
+	Whitelist    json.RawMessage `json:"whitelist"`
+}
+
+type ArgsTWAP struct {
+	EAttest   json.RawMessage `json:"eAttest"`
+	TAttest   json.RawMessage `json:"tAttest"`
+	Whitelist json.RawMessage `json:"whitelist"`
+}
+
+type Result struct {
+	Success bool
+	Error   string
+}
+
+type PriceSamples []price.Price
+
+type ResultPriceSamples struct {
+	Result       Result
+	PriceSamples PriceSamples
+}
+
+type ResultTWAP struct {
+	Result Result
+	TWAP   float64
+}
+
+func extractPriceSamples(
+	eAttest,
+	tAttest,
+	whitelist json.RawMessage,
+) (
+	PriceSamples,
+	error,
+) {
+	// bootstrap with empty samples if we don't have a transitive attestation
+	if tAttest == nil {
+		return PriceSamples{}, nil
+	}
+
+	verifyOut, err := as.VerifyAttestation(
```

Comment:
This is probably fine for now, but I think we may want to avoiding this pattern so high up and maybe improve the types a little in here so that it isn't just passing around byte arrays (at the very least, they could be named types that are actually byte arrays).

Reply:
I agree that this could be locked down a bit. I'll leave as is for now - we should address this when looking to improve `verify.go`